contrib/datadog/tracing/interceptor.go

```diff
@@ -0,0 +1,200 @@
+package tracing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+var (
+	_ interceptor.Tracer     = tracerImpl{}
+	_ interceptor.TracerSpan = &tracerSpan{}
+)
```

Comment:
I will remove them, it's a very personal thing of mine to show that a given struct is implementing a given interface

Reply:
It's a common thing for many. What I mean about these not being needed is not that you don't want this assertion, it's that the assertion is done by `return tracerImpl{` and `return &tracerSpan{`.

(unresolving because adding comments on resolved threads are often not read)